packages/web-components/src/keplr-connection/KeplrConnection.js

```diff
@@ -12,17 +13,28 @@ const DEFAULT_NETWORK_CONFIG = 'https://main.agoric.net/network-config';
 export const makeAgoricKeplrConnection = async (
   networkConfig = DEFAULT_NETWORK_CONFIG,
   context = makeImportContext(),
+  chainStorageWatcher,
 ) => {
   const { chainId, rpcs } = await getChainInfo(networkConfig);
   const address = await getKeplrAddress(chainId);
 
-  const leader = makeLeader(networkConfig);
-  const walletNotifiers = await watchWallet(leader, address, context, rpcs);
+  let walletNotifiers;
+  let leader;
+  if (chainStorageWatcher) {
+    walletNotifiers = await watchWalletBatched(
+      chainStorageWatcher,
+      address,
+      rpcs[Math.floor(Math.random() * rpcs.length)],
+    );
+  } else {
+    leader = makeLeader(networkConfig);
+    walletNotifiers = await watchWallet(leader, address, context, rpcs);
+  }
 
   return {
     address,
     chainId,
-    unserializer: context.fromBoard,
+    unserializer: chainStorageWatcher ? context.fromBoard : undefined,
```

Comment:
this is also a breaking change.  it also raises the question of whether this is necessary.